### PR TITLE
chore(sqllab): Relocate get bootstrap data logic

### PR DESCRIPTION
**Pull request by justinpark**
### SUMMARY
Part 1 of SQLLab SPA migration: [detail link](https://docs.google.com/document/d/1opyNuaw-CEAD9tG6fx_zqNcZwe3_soQgsnSJTP4mQkk/edit?usp=sharing)

This commit relocates the existing `_get_sqllab_tabs` to the utils in sqllab in order to share the same logic in the sqllab api v1 endpoint.
This commit also refactors the `defaultDBId` by common config parameters which is less relevant to sqlLab root state.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
N/A

### TESTING INSTRUCTIONS
Go to SQLLab and no errors shown

### ADDITIONAL INFORMATION
- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by justinpark** [comment]
cc: @EugeneTorap

**Comment by EugeneTorap** [comment]
Hi @michael-s-molina @john-bodley @villebro! Can you help with this PR?

File: superset/sqllab/utils.py [hunk 1]
```diff
@@ -14,11 +14,29 @@
 # KIND, either express or implied.  See the License for the
 # specific language governing permissions and limitations
 # under the License.
-from typing import Any
+from typing import Any, Union
```
Comment (michael-s-molina) [inline superset/sqllab/utils.py:17]:
```suggestion
from __future__ import annotations

from typing import Any
```

File: superset/sqllab/utils.py [hunk 2]
```diff
@@ -56,3 +74,46 @@ def write_ipc_buffer(table: pa.Table) -> pa.Buffer:
         writer.write_table(table)
 
     return sink.getvalue()
+
+
+def bootstrap_sqllab_data(user_id: Union[int, None]) -> dict[str, Any]:
```
Comment (michael-s-molina) [inline superset/sqllab/utils.py:79]: @justinpark you can use `int | None` but you need to add `from __future__ import annotations` as the first import.
```suggestion
def bootstrap_sqllab_data(user_id: int | None) -> dict[str, Any]:
```

**Comment by michael-s-molina** [comment]
To fix the Python Misc CI error, you need to fix the order of the imports in these files:

- `superset/views/core.py`
- `superset/tests/integration_tests/core_tests.py`

To fix the import order, just run the `pre-commit` hook.

**Review by michael-s-molina — APPROVED** [review]
LGTM. Thank you @justinpark!

**Comment by EugeneTorap** [comment]
Can we merge it?

**Pull request merged**